lib/Wikia/src/Util/Consul/ConsulService.php

```diff
@@ -0,0 +1,62 @@
+<?php
+namespace Wikia\Util\Consul;
+
+class ConsulService {
+
+	protected $config;
+
+	public function __construct( ConsulConfig $config ) {
+		$this->config = $config;
+		return $this;
+	}
+
+	/**
+	 * @return ConsulConfig
+	 */
+	public function getConfig() {
+		return $this->config;
+	}
+
+	/**
+	 * @param ConsulConfig $config
+	 */
+	public function setConfig( ConsulConfig $config ) {
+		$this->config = $config;
+	}
+
+
+	/**
+	 * Get all host/port for the service.
+	 * @param $consulServiceHost
+	 * @return array
+	 */
+	public function getResolvedRecords( $consulServiceHost ) {
+		$records = $this->doResolve( $consulServiceHost );
+		$resolved = [ ];
+		if ( is_array( $records ) && !empty( $records ) ) {
+			foreach ( $records as $record ) {
+				$resolved[ ] = [ 'host' => $record[ 'target' ], 'port' => $record[ 'port' ] ];
+			}
+		}
+		return $resolved;
+	}
+
+	protected function doResolve( $consulServiceHost ) {
```

Comment:
This is only called once, and it's protected, so I'm not sure if we need a helper function here.  Not every PHP core function call needs a wrapper.